lib/hashie/mash.rb

```diff
@@ -209,7 +209,7 @@ def deep_merge(other_hash, &blk)
     def deep_update(other_hash, &blk)
       other_hash.each_pair do |k, v|
         key = convert_key(k)
-        if regular_reader(key).is_a?(Mash) && v.is_a?(::Hash)
+        if key?(key) && regular_reader(key).is_a?(Mash) && v.is_a?(::Hash)
```

Comment:
Simple fix! Since I'm looking at the line, I feel like we should swap the order of the `v.is_a?` check and the reader check, but that's not something we need to address in this PR.

Reply:
+1 @michaelherold 

@laertispappas I know this is outside of the scope here, but care to make this change so we don't forget? It's a lot faster to do `is_a?` than to invoke `regular_reader`. Thanks.